File: manual/data/java/tests/test_seaborn_set_theme.java
package com.example;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.assertEquals;

public class test_seaborn_set_theme {

    @Test
    public void test_default_theme() {
        function_seaborn_set_theme themeSetter = new function_seaborn_set_theme();
        String result = themeSetter.set_theme("darkgrid", "notebook", "deep");
        assertEquals("{'style': 'darkgrid', 'context': 'notebook', 'palette': 'deep'}", result);
    }

    @Test
    public void test_custom_theme() {
        function_seaborn_set_theme themeSetter = new function_seaborn_set_theme();
        String result = themeSetter.set_theme("white", "talk", "muted");
        assertEquals("{'style': 'white', 'context': 'talk', 'palette': 'muted'}", result);
    }

    @Test
    public void test_empty_style() {
        function_seaborn_set_theme themeSetter = new function_seaborn_set_theme();
        String result = themeSetter.set_theme("", "notebook", "deep");
        assertEquals("{'style': 'darkgrid', 'context': 'notebook', 'palette': 'deep'}", result);
    }

    @Test
    public void test_empty_context() {
        function_seaborn_set_theme themeSetter = new function_seaborn_set_theme();
        String result = themeSetter.set_theme("darkgrid", "", "deep");
        assertEquals("{'style': 'darkgrid', 'context': 'notebook', 'palette': 'deep'}", result);
    }

    @Test
    public void test_empty_palette() {
        function_seaborn_set_theme themeSetter = new function_seaborn_set_theme();
        String result = themeSetter.set_theme("darkgrid", "notebook", "");
        assertEquals("{'style': 'darkgrid', 'context': 'notebook', 'palette': 'deep'}", result);
    }
}

File: manual/code/test_suites/python/python/test_seaborn_set_theme.py
import unittest
from function_seaborn_set_theme import SeabornThemeSetter

class TestSeabornThemeSetter(unittest.TestCase):
    def test_default_theme(self):
        theme_setter = SeabornThemeSetter()
        result = theme_setter.set_theme()
        self.assertEqual(result, "{'style': 'darkgrid', 'context': 'notebook', 'palette': 'deep'}")

    def test_custom_theme(self):
        theme_setter = SeabornThemeSetter()
        result = theme_setter.set_theme(style="white", context="talk", palette="muted")
        self.assertEqual(result, "{'style': 'white', 'context': 'talk', 'palette': 'muted'}")

    def test_empty_style(self):
        theme_setter = SeabornThemeSetter()
        result = theme_setter.set_theme(style="")
        self.assertEqual(result, "{'style': 'darkgrid', 'context': 'notebook', 'palette': 'deep'}")

    def test_empty_context(self):
        theme_setter = SeabornThemeSetter()
        result = theme_setter.set_theme(context="")
        self.assertEqual(result, "{'style': 'darkgrid', 'context': 'notebook', 'palette': 'deep'}")

    def test_empty_palette(self):
        theme_setter = SeabornThemeSetter()
        result = theme_setter.set_theme(palette="")
        self.assertEqual(result, "{'style': 'darkgrid', 'context': 'notebook', 'palette': 'deep'}")

if __name__ == "__main__":
    unittest.main()
